### PR TITLE
fix(OnyxNavBar): align horizontal padding with grid

### LINKED ISSUE #1279: Fix nav bar spacings not aligned with grid
The nav bar left/right spacings are not always aligned with the grid / page content on all breakpoints, see [this playground reproduction](https://playground.onyx.schwarz/#eNp9UsFOwzAM/ZUoXNf2AKdRkBjaAQTbBFyQIqGo9bqMNomStGya+u84abt10rRb/Pz8/GznQJ+0jpsa6JSmNjNCO1JyWTww6iyjxIKr9SOTotLKOHIgS7nfYckb36vaTUK44gWM4wVvZtyQlqyNqgijVrhIIc7oPZNp0nVBTQwcVLrkDjAiJD3TDhCCA4XcyKDb4z2978W1jiSvAG2/732E1pNBIRl16aFz2yfJXDQkK7m1KOQtR4UReZQp6biQYBh9/Fa1IRpLiUdBOrIBA3EcpwkWH1teaNCB4/lGzugE942Ka1HEW6sknuPgixjNVKVFCWapnVASbzIlIeNzvCzV32vAnKlhMuDZBrLfC/jW4hGm+FgZsGAaYPSYc9wU4Lr0/HMBO3wfk5XK6xLZV5IfYFVZe48dbVbLHG2PeMHtS/hIQhZfdr7D/dlhKG/UM9vAZxT/5POV0U92b+O7UMdki1v8acB4TVygX6x1tP0HmYr25w==)

Might potential be fixed by #1252

### PR DESCRIPTION
**Pull request by larsrickert**
closes #1279

- align nav bar horizontal padding with grid

## Checklist

- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: packages/sit-onyx/playwright/snapshots/components/OnyxNavBar/grid-default-md-firefox-linux.png
Comment (BoppLi) [inline packages/sit-onyx/playwright/snapshots/components/OnyxNavBar/grid-default-md-firefox-linux.png:1]: the examples look fine in the screenshots but in http://localhost:5173/grid it is misaligned e.g. on a screenwidth of 970px:
image

The nav has 64px and the body 32px, is this a bug of the demo app?

File: .changeset/many-zoos-count.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"sit-onyx": patch
+---
+
+fix(OnyxNavBar): align vertical padding with grid
```
Comment (BoppLi) [inline .changeset/many-zoos-count.md:5]: left/right would be
```suggestion
fix(OnyxNavBar): align horizontal padding with grid
```

File: packages/sit-onyx/playwright/snapshots/components/OnyxNavBar/grid-default-md-firefox-linux.png
Comment (larsrickert) [reply packages/sit-onyx/playwright/snapshots/components/OnyxNavBar/grid-default-md-firefox-linux.png:1]: It looks fine for me, did you restart the demo app?

image

Comment (BoppLi) [reply packages/sit-onyx/playwright/snapshots/components/OnyxNavBar/grid-default-md-firefox-linux.png:1]: weird, now it's fine.. must have been a glitch ¯\_(ツ)_/¯

File: packages/sit-onyx/src/components/OnyxNavBar/OnyxNavBar.vue [hunk 2]
```diff
@@ -195,7 +194,7 @@ $height: 3.5rem;
       align-items: center;
       gap: $gap;
       height: 100%;
-      padding-inline: var(--onyx-spacing-3xl);
+      padding-inline: var(--onyx-grid-margin);
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxNavBar/OnyxNavBar.vue:197]: question: will this variable be defined even if the nav bar is used without a grid layout surrounding it?

**Review by BoppLi — APPROVED** [review]

File: packages/sit-onyx/src/components/OnyxNavBar/OnyxNavBar.vue (hunk 2 above)
Comment (larsrickert) [reply packages/sit-onyx/src/components/OnyxNavBar/OnyxNavBar.vue:197]: Yes that’s a global variable which is always defined

**Pull request merged**